### PR TITLE
add tests for websitePaymentsModel

### PR DESCRIPTION
**Pull request by YaroslavSemeniuk**
add tests for websitePaymentsModel

### CODE REVIEW

File: test/models/WebsitePaymentsModel/test.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    let host, condition, sort, websitePayments;
+    beforeEach(async () => {
+      await dropDatabase();
+      host = faker.random.string();
+      condition = { host };
+      sort = { createAt: 1 };
+
+      const websitePayment1 = {
+        host,
+        type: PAYMENT_TYPES.WRITE_OFF,
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:18]: why do you need different payment types?

File: test/models/WebsitePaymentsModel/test.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        amount: 10,
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:19]: _.random()

File: test/models/WebsitePaymentsModel/test.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+const { PAYMENT_TYPES } = require('../../../constants/sitesConstants');
+
+describe('WebsitePaymentsModel', () => {
+  describe('On find', () => {
+    let host, condition, sort, websitePayments;
+    beforeEach(async () => {
+      await dropDatabase();
+      host = faker.random.string();
+      condition = { host };
+      sort = { createAt: 1 };
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:14]: You will test  only ascending sort?

File: test/models/WebsitePaymentsModel/test.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        host,
+        type: PAYMENT_TYPES.WRITE_OFF,
+        amount: 20,
+      };
+      const websitePayment4 = {
+        type: PAYMENT_TYPES.REFUND,
+        amount: 13,
+      };
+      websitePayments = [websitePayment1, websitePayment2, websitePayment3, websitePayment4];
+      for (const websitePayment of websitePayments) {
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:36]: for i with random counter for known host and one more loop for unknown

File: test/models/WebsitePaymentsModel/test.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+      expect(result).to.have.length(3);
+    });
+    it('Should return array without one element and when searching not to find it', async () => {
+      const { result } = await WebsitePaymentsModel.find({ condition, sort });
+      expect(result).to.not.include(websitePayments[3]);
+    });
+    it('Should return array whose first element will contain the smallest amount & the last largest', async () => {
+      const { result } = await WebsitePaymentsModel.find({ condition, sort });
+      expect(result[0].amount).to.be.eq(websitePayments[0].amount);
+      expect(result[2].amount).to.be.eq(websitePayments[2].amount);
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:52]: 1)2 expect in one it
2)check first element to be < last element

File: test/models/WebsitePaymentsModel/test.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    });
+    it('Should check that the result will be undefined', async () => {
+      const { error } = await WebsitePaymentsModel.find(faker.random.string());
+      expect(error).to.be.undefined;
+    });
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:57]: what are you testing here?
add test error to be exist
add sort descending test
add host check for result array

File: test/models/WebsitePaymentsModel/test.js [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+  describe('On aggregate', () => {
+    let websitePayments, condition, name;
+    beforeEach(async () => {
+      await dropDatabase();
+      name = faker.name.firstName();
+      condition = [{ $match: { userName: name } }];
+      const websitePayment1 = { name };
+      const websitePayment2 = { name };
+      const websitePayment3 = { name: faker.name.firstName() };
+      const websitePayment4 = { name: faker.name.firstName() };
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:69]: duplicate code
you don't need those variables

File: test/models/WebsitePaymentsModel/test.js [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+    it('Should return an array in which the names match the condition', async () => {
+      const { result } = await WebsitePaymentsModel.aggregate(condition);
+      expect(result).to.have.length(2);
+    });
+    it('Should return an array that does not contain elements that satisfy the condition', async () => {
+      const { result } = await WebsitePaymentsModel.aggregate(condition);
+      expect(result).to.not.include(websitePayments[2]);
+      expect(result).to.not.include(websitePayments[3]);
+    });
+  });
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:84]: 1)better be check that every result that you got has proper userName
2) 2 expect

File: test/models/WebsitePaymentsModel/test.js [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+      name = faker.name.firstName();
+      condition = { userName: name };
+      const websitePayment1 = { name };
+      const websitePayment2 = { name: faker.name.firstName() };
+      const websitePayment3 = { name: faker.name.firstName() };
+      const websitePayment4 = { name: faker.name.firstName() };
+      websitePayments = [websitePayment1, websitePayment2, websitePayment3, websitePayment4];
+      for (const websitePayment of websitePayments) {
+        await WebsitePaymentsFactory.Create({ ...websitePayment });
+      }
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:99]: same as previous

File: test/models/WebsitePaymentsModel/test.js [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+    });
+    it('Should return one element that satisfies the condition', async () => {
+      const { result } = await WebsitePaymentsModel.findOne(condition);
+      expect(result.userName).to.be.eq(websitePayments[0].name);
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:103]: why don't you compare with a variable name?

File: test/models/WebsitePaymentsModel/test.js [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+      name1 = faker.name.firstName();
+      name2 = faker.name.firstName();
+      const websitePayment1 = { name: name1, amount: 15 };
+      const websitePayment2 = { name: name1, amount: 12 };
+      const websitePayment3 = { name: name2, amount: 15 };
+      const websitePayment4 = { amount: 13 };
+      websitePayments = [websitePayment1, websitePayment2, websitePayment3, websitePayment4];
+      for (const websitePayment of websitePayments) {
+        await WebsitePaymentsFactory.Create({ ...websitePayment });
+      }
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:126]: same

File: test/models/WebsitePaymentsModel/test.js [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+    });
+    it('Should return an array of two unique elements', async () => {
+      const { result } = await WebsitePaymentsModel.distinct({ field, query });
+      expect(result).to.have.members([websitePayments[2].name, websitePayments[0].name]);
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:130]: name1, name2

File: test/models/WebsitePaymentsModel/test.js [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+const { WebsitePaymentsFactory } = require('test/factories');
+const { PAYMENT_TYPES } = require('../../../constants/sitesConstants');
+
+describe('WebsitePaymentsModel', () => {
+  describe('On find', () => {
+    let host, condition, sort, websitePayments;
+    beforeEach(async () => {
+      await dropDatabase();
+      host = faker.random.string();
+      condition = { host };
```
Comment (Fesmofet) [inline test/models/WebsitePaymentsModel/test.js:13]: condition may change better write it in method params directly

**Review by Waiviogit — APPROVED** [review]

**Pull request merged**